app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java

```diff
@@ -168,8 +175,23 @@ private void startSensorOrAskForG6Code() {
         final int cap = 20;
         if (Ob1G5CollectionService.usingCollector() && Ob1G5StateMachine.usingG6()) {
             if (JoH.pratelimit("dex-stop-start", cap)) {
+                int TX_dys = DexTimeKeeper.getTransmitterAgeInDays(getTransmitterID());
+                PersistentStore.setBoolean(TX_EOL, false);
```

Comment:
Don't use persistent store for temporary data. If you don't want to pass parameters then you can use the fast store like `FastStore.getInstance().putL("blah",1);` and retrieve with `FastStore.getInstance().get("blah");`

Reply:
Thanks for teaching me yet one more thing.  FastStore?  Awesome  

My concern with this PR was something else though.  May be I was wrong.
I thought I was supposed to use the same variables from different classes.  So, I thought this was a mistake I had made:
https://github.com/NightscoutFoundation/xDrip/pull/2160

I already have a case that creates different branches depending on the number of days.  I thought I was supposed to do that only once and then use the same variables everywhere instead of having the case statement over and over.
So, I thought may be I was supposed to go back to that PR and make changes to it such that I can use the same variables.

Are you saying that if I make the change you have suggested here, you are OK with everything else?

Reply:
You can pass the variables to the dialog method if that makes more sense than using `FastStore`

You want to avoid duplicating code so if you evaluate variables once then where possible reuse these.

You could pass a Runnable to your dialog method and create a callback like that. I'm sure how complicated it needs to be.

I don't find this PR that easy to read but it looks like it should work and implement the desired functionality. I just don't want `PersistentStore` used for temporary data because there is an overhead with that.